osi_trafficcommand.proto

```diff
@@ -0,0 +1,507 @@
+
+    // The data timestamp of the simulation environment. Zero time is arbitrary
+    // but must be identical for all messages. Zero time does not need to
+    // coincide with the UNIX epoch. Recommended is the starting time point of
+    // the simulation.
+    //
+    // \note For traffic command data the timestamp coincides both with
+    // the notional simulation time the data applies to and the time it was sent
+    // (there is no inherent latency for traffic command data, as opposed
+    // to sensor data).
+    //
+    optional Timestamp timestamp = 2;
+
+    // The ID of this traffic participant.
+    //
+    optional Identifier traffic_participant_id = 3;
+
+    // Commanded Traffic Action(s) if any
+    //
+    // \note OSI uses singular instead of plural for repeated field names.
+    //
+    // \note If more than one action is being supplied in one command message
+    // all actions are executed in parallel.
+    //
+    repeated TrafficAction action = 4;
+
+}
+
+//
+// \brief Atomic Traffic Actions to be performed
+//
+// The actual commands being transmitted are the atomic traffic actions
+// described by this message.
+//
+// \note This message is notionally a multiple choice selection, i.e. only 
+// certain combinations of atomic traffic actions shall be transmitted within 
+// certain time intervals (e.g. for plausibity reasons). The restrictions 
+// regarding that are not part of this message, yet are seen as a task of the 
+// scenario description (e.g. OpenSCENARIO). 
+//
+// \note All TrafficActions are expected to be sent only once, to indicate
+// that they must start immediately. This is also true, if their execution 
+// is expected to take simulation time. To inform the traffic participant 
+// model that certain actions must or shall be terminated, there are 
+// explicit Actions nested inside this message (AbortActionsAction, 
+// EndActionsAction), which hold a reference to the respective actions.
+//
+message TrafficAction
+{
+    // A FollowTrajectoryAction
+    //
+    optional FollowTrajectoryAction follow_trajectory_action = 1;
+
+    // A FollowPathAction
+    //
+    optional FollowPathAction follow_path_action = 2;
+
+    // An AcquireGlobalPositionAction
+    //
+    optional AcquireGlobalPositionAction acquire_global_position_action = 3;
+
+    // A LaneChangeAction
+    //
+    optional LaneChangeAction lane_change_action = 4; 
+
+    // A SpeedAction
+    //
+    optional SpeedAction speed_action = 5;    
+
+    // An AbortActionsAction
+    //
+    optional AbortActionsAction abort_actions_action = 6; 
+
+    // An EndActionsAction
+    //
+    optional EndActionsAction end_actions_action = 7; 
+}
+
+//
+// \brief The StatePoint definition
+//
+// \note The StatePoint definition does not define mandatory fields.
+// The context defines how and what fields are used.
+// For example: FollowPathAction does not use timestamp in contrast to
+// FollowTrajectoryAction. 
+//
+message StatePoint
```

Comment:
As discussed with @caspar-ai and @Krishna626 the SteatePoint definition might be a candidate for osi_common.proto (which is imported by this message), because the definition could be re-used in other messages (e.g. TrafficUpdate). TBD prior merging.

Reply:
It depends: For the actual state update in TrafficUpdate more fields are needed. For reuse as the future trajectory points as proposed by @caspar-ai I think this makes potential sense. We could however do the move in the pull request that adds the future trajectory points, since it already is a global message that should not break anything...

Reply:
Yes, that's what @caspar-ai did here https://github.com/OpenSimulationInterface/open-simulation-interface/pull/401#issuecomment-722533543. So I mark this conversation as resolved.